src/azure-cli/azure/cli/command_modules/iot/_help.py

```diff
@@ -65,6 +65,9 @@
   - name: Update shared access policy 'MyPolicy' in an Azure IoT Hub Device Provisioning Service instance with EnrollmentWrite right
     text: >
         az iot dps policy update --dps-name MyDps --resource-group MyResourceGroup --policy-name MyPolicy --rights EnrollmentWrite
+  - name: Regenerate keys for access policy 'MyPolicy' by updating keys to empty values
+    text: >
+        az iot dps policy update --dps-name MyDps --resource-group MyResourceGroup --policy-name MyPolicy --primary-key "" --secondary-key ""
```

Comment:
are we trying to remove or renew keys here? cause if renew then wouldn't naming like `--renew-primary` or whatever be better?

there should be some key renewal stuff in iot hub

Reply:
IoT Hub control plane has a dedicated `policy renew-key` command, but in this case it is around enabling the more generic `dps policy update` to allow service renewal of keys.

Reply:
we could also make the `--primary-key` value optional so if a person just specifies `--primary-key` then the service will regenerate it for them (rather than awkwardly adding in "")

Reply:
@vilit1 The way it's currently coded I don't see a good way to differentiate between an empty value: `--primary-key` and not passing the arg at all. If update chooses to accept `None` as a value - it will then regenerate the keys on every update if you don't provide them to the command. I think for now it's ok to use the empty string as a way to say "I am providing this value but I want it to be empty" as opposed to "I have not provided this value and don't want it to change".